tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/metrics/BulkheadMetricTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.fault.tolerance.tck.metrics;
+
+import static org.eclipse.microprofile.fault.tolerance.tck.metrics.Exceptions.expectBulkheadException;
+import static org.eclipse.microprofile.fault.tolerance.tck.metrics.MetricComparator.approxMillis;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.metrics.Histogram;
+import org.eclipse.microprofile.metrics.Snapshot;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+
+public class BulkheadMetricTest extends Arquillian {
+
+    public static WebArchive deploy() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "ftMetricBulkhead.war")
+                .addClasses(BulkheadMetricBean.class, MetricGetter.class);
+        return war;
+    }
+    
+    @Inject private BulkheadMetricBean bulkheadBean;
+    @Inject private AsyncCaller async;
+    
+    @Test
+    public void bulkheadMetricTest() throws InterruptedException, ExecutionException {
+        MetricGetter m = new MetricGetter(BulkheadMetricBean.class, "waitFor");
+        m.baselineCounters();
+        
+        Future<?> f1 = async.run(() -> bulkheadBean.waitFor(3000));
+        Future<?> f2 = async.run(() -> bulkheadBean.waitFor(3000));
+        
+        assertThat("concurrent executions", m.getBulkheadConcurrentExecutions().get(), is(2L));
+        
+        f1.get();
```

Comment:
If waitFor is rewritten using a completable future, it should complete the future passed to waitFor earlier